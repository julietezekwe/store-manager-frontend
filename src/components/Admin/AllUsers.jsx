import React, { Component } from 'react';
import { connect } from 'react-redux';
import Form from '../common/Form.jsx';
import { getAllUsers } from '../../actions/usersAction';

export class AllUsers extends Component {
  async componentDidMount() {
    await this.props.getAllUsers();
  }

  render() {
    const { users, isLoadingUsers } = this.props.login;

    return (
      <React.Fragment>
        <div className="col-8">
          <Form action="addUser" />
          <div className="card col-11">
            <table className="table">
              <thead>
                <tr>
                  <th>Name</th>
                  <th>Username</th>
                  <th>Email</th>
                  <th>Role</th>
                  <th>Action</th>
                </tr>
              </thead>
              <tbody>
                {!isLoadingUsers
                  && users.UsersModel.map(user => (
                    <tr key={user.id}>
                      <td>{user.name}</td>
                      <td>{user.username}</td>
                      <td>{user.email}</td>
                      <td>{user.role}</td>
                      <td>
                        <Form action="eidtUser" userId={user.id} /> &nbsp;
                        &nbsp;
                        <Form action="deleteUser" userId={user.id} />
                      </td>
                    </tr>
                  ))}
              </tbody>
            </table>
          </div>
        </div>
      </React.Fragment>
    );
  }
}
const mapStateToProps = state => ({
  login: state.login,
  users: state.login.users,
});
export default connect(
  mapStateToProps,
  { getAllUsers },
)(AllUsers);
