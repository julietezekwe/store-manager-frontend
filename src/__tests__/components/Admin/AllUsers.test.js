/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import { AllUsers } from '../../../components/Admin/AllUsers.jsx';

let wrapper;
const props = {
  loginUser: () => Promise.resolve(),
  login: {
    isLoadingUsers: false,
    users: {
      UsersModel: [{
        id: 1,
        name: 'admin',
        username: 'admin',
        email: 'admin@gmail.com',
        role: 'admin',
      }],
    },
  },
};
beforeEach(() => {
  wrapper = shallow(<AllUsers {...props} />);
});
describe('App Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
