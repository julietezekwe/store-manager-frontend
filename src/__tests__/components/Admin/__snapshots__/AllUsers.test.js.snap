// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
<Fragment>
  <div
    className="col-8"
  >
    <Connect(Form)
      action="addUser"
    />
    <div
      className="card col-11"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              Username
            </th>
            <th>
              Email
            </th>
            <th>
              Role
            </th>
            <th>
              Action
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            key="1"
          >
            <td>
              admin
            </td>
            <td>
              admin
            </td>
            <td>
              admin@gmail.com
            </td>
            <td>
              admin
            </td>
            <td>
              <Connect(Form)
                action="eidtUser"
                userId={1}
              />
                  
              <Connect(Form)
                action="deleteUser"
                userId={1}
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</Fragment>
`;
